flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -169,9 +169,8 @@ void retriggerSubpartitionRequest(int subpartitionIndex) throws IOException {
 
 	@Override
 	Optional<BufferAndAvailability> getNextBuffer() throws IOException {
-		checkState(partitionRequestClient != null, "Queried for a buffer before requesting a queue.");
-
 		checkError();
+		checkState(partitionRequestClient != null, "Queried for a buffer before requesting a queue.");
```

Comment:
BTW, is the message "before requesting a queue" right? Because we're actually requesting a partition. I'm not sure whether we use this message, maybe some historical reasons from non-credit-based network model?

Reply:
I think it's mostly fine. If you want, the message could be updated to:
> Queried for a buffer before requesting a subpartition